dotrix_core/src/renderer.rs

```diff
@@ -178,7 +185,8 @@ pub fn bind(mut renderer: Mut<Renderer>, mut assets: Mut<Assets>) {
 
     let mut loaded = true;
 
-    for (_id, shader) in assets.iter_mut::<Shader>() {
+    for (id, shader) in assets.iter_mut::<Shader>() {
+        renderer.drop_pipeline(*id);
```

Comment:
If shaders are being loaded from files, it may easily happen, that some shaders will be already loaded, while others will be still in process of loading from file system. In that case, this call will drop newly loaded pipelines for shaders that are ready. And it will occur on every frame until all shaders are loaded.

I suggest to remove `renderer.drop_pipeline(*id);` from here and add `drop_all_pipelines()` method.

We can also consider to rename the `reload()` method, or keep the name and call `drop_all_pipelines()` inside of it.

Reply:
I see I was wondering if this was a likely case

Reply:
What about calling `drop_all_pipelines` in `reload`? When is `reload` called would that be an issue?

Reply:
Ok I added the `drop_all_pipelines` and called it from `reload`